desc/compute/_stability.py

```diff
@@ -232,3 +232,388 @@ def _magnetic_well(params, transforms, profiles, data, **kwargs):
+    "Default 15 points linearly spaced in [-π/2,π/2]",
+)
+def _ideal_ballooning_gamma2(params, transforms, profiles, data, **kwargs):
+    """
+    Ideal-ballooning growth rate finder.
+
+    This function uses a finite-difference method
+    to calculate the maximum growth rate against the
+    infinite-n ideal ballooning mode. The equation being solved is
+
+    d/dζ(g dX/dζ) + c X = γ f X, g, f > 0
+
+    where
+
+    𝛋 = 𝐛 ⋅∇ 𝐛
+    g = a_N^3 * B_N * (b ⋅∇ζ) * |∇α|², / B,
+    c = a_N/B_N * (1/ b ⋅∇ζ) * dψ/dρ * dp/dψ * (b × 𝛋) ⋅|∇α|/ B**2,
+    f = a_N * B_N^3 *|∇α|² / B^3 * (1/ b ⋅∇ζ) ,
+    are needed along a field line to solve the ballooning equation once.
+
+    To obtain the parameters g, c, and f, we need a set of parameters
+    provided in the list ``data`` above. Here's a description of
+    these parameters:
+
+    - a: minor radius of the device
+    - g^aa: |grad alpha|^2, field line bending term
+    - g^ra: (grad alpha dot grad rho) integrated local shear
+    - g^rr: |grad rho|^2 flux expansion term
+    - cvdrift: geometric factor of the curvature drift
+    - cvdrift0: geoetric factor of curvature drift 2
+    - |B|: magnitude of the magnetic field
+    - B^zeta: inverse of the jacobian
+    - p_r: dp/drho, pressure gradient
+    - phi: coordinate describing the position in the toroidal angle
+    along a field line
+    """
+    source_grid = transforms["grid"].source_grid
+    # Vectorize in rho later
+    rho = source_grid.meshgrid_reshape(data["rho"], "arz")
+
+    psi_b = params["Psi"] / (2 * jnp.pi)
+    a_N = data["a"]
+    B_N = 2 * psi_b / a_N**2
+
+    zeta0 = kwargs.get("zeta0", jnp.linspace(-0.5 * jnp.pi, 0.5 * jnp.pi, 15))
+    N_zeta0 = len(zeta0)
+
+    # This would fail with rho vectorization
+    iota = jnp.mean(data["iota"])
+    shear = jnp.mean(data["shear"])
+    psi = jnp.mean(data["psi"])
+    sign_psi = jnp.sign(psi)
+    sign_iota = jnp.sign(iota)
+
+    N_alpha = int(source_grid.num_alpha)
+
+    # phi is the same for each alpha
+    phi = source_grid.nodes[::N_alpha, 2]
+    N_zeta = len(phi)
+
+    B = source_grid.meshgrid_reshape(data["|B|"], "arz")
+    B_sup_zeta = source_grid.meshgrid_reshape(data["B^zeta"], "arz")
+    gradpar = B_sup_zeta / B
+
+    # This would fail with rho vectorization
+    dpdpsi = jnp.mean(mu_0 * data["p_r"] / data["psi_r"])
+
+    g_sup_aa = source_grid.meshgrid_reshape(data["g^aa"], "arz")[None, ...]
+    g_sup_ra = source_grid.meshgrid_reshape(data["g^ra"], "arz")[None, ...]
+    g_sup_rr = source_grid.meshgrid_reshape(data["g^rr"], "arz")[None, ...]
+
+    gds2 = jnp.reshape(
+        rho**2
+        * (
+            g_sup_aa
+            - 2 * sign_iota * shear / rho * zeta0[:, None, None, None] * g_sup_ra
+            + zeta0[:, None, None, None] ** 2 * (shear / rho) ** 2 * g_sup_rr
+        ),
+        (N_alpha, N_zeta0, N_zeta),
+    )
+
+    f = a_N**3 * B_N * gds2 / B**3 * 1 / gradpar
+    g = a_N**3 * B_N * gds2 / B * gradpar
```

Comment:
This is dimensionless. just noting

Reply:
Yes, g and c are dimensionless but f is dimensional. I wanted all three terms to have the same scaling because it made the eigenvalue solver easier (eigenvalue is ~ 1 rather that 1e-4).  As a consequence, lambda becomes dimensional which we'll have to justify somehow.

Reply:
So I have corrected it now. g, c, and f are each normalized.